### PR TITLE
Enrich DB error handling

### PR DESCRIPTION
**Pull request by PetarTodorovv**
**Description**

Changes proposed in this pull request:
Add new cases to enrich DB error handling
Fix foreign key error handling

### CODE REVIEW

File: components/director/pkg/resource/resource.go [hunk 1]
```diff
@@ -20,3 +20,15 @@ const (
 	AutomaticScenarioAssigment Type = "AutomaticScenarioAssigment"
 	Webhook                    Type = "Webhook"
 )
+
+type SQLOperation string
```
Comment (DimitarPetrov) [inline components/director/pkg/resource/resource.go:24]: This type and the constants below belong more to the persistence/const.go

**Review by DimitarPetrov — APPROVED** [review]

File: components/director/pkg/resource/resource.go (hunk 1 above)
Comment (PetarTodorovv) [reply components/director/pkg/resource/resource.go:24]: discussed offline

**Review by DimitarPetrov — APPROVED** [review]

**Review by DimitarPetrov — APPROVED** [review]

**Pull request merged**